static/js/hotspots.js

```diff
@@ -2,41 +2,183 @@ var hotspots = (function () {
 
 var exports = {};
 
-exports.show = function (hotspot_list) {
-    $('.hotspot').hide();
-    for (var i = 0; i < hotspot_list.length; i += 1) {
-        $("#hotspot_".concat(hotspot_list[i].name)).show();
-    }
+var HOTSPOT_LOCATIONS = {
+    click_to_reply: {
+        element: '.selected_message .messagebox-content',
+        icon: {
+            top: -71,
+            left: 284,
+        },
+        popover: {
+            top: -210,
+            left: -125,
+            arrow_placement: 'bottom',
+        },
+    },
+    new_topic_button: {
+        element: '#left_bar_compose_stream_button_big',
+        icon: {
+            top: -8,
+            left: -3,
+        },
+        popover: {
+            top: -213,
+            left: -90,
+            arrow_placement: 'bottom',
+        },
+    },
+    stream_settings: {
+        element: '#join_unsub_stream',
+        icon: {
+            top: 7,
+            left: 158.7,
+        },
+        popover: {
+            top: 27,
+            left: -124,
+            arrow_placement: 'top',
+        },
+    },
 };
```

Comment:
The numbers are offset values for absolute positioning. It works alright for responsive UI, as the hotspots remain positioned relative to the elements they highlight. An alternative could be to specify the location on the element (i.e. for icons, `top-left`, `top-right`, `bottom-left`, `bottom-right`, and `center`; for popovers just `top`, `left`, `bottom`, and `right`) and use element height and width to compute placement. I think it would be slightly easier to maintain, though not as visually precise.